plugin/entrySchema.go

```diff
@@ -123,6 +132,12 @@ func NewEntrySchema(e Entry, label string) *EntrySchema {
 // Note that UnmarshalJSON is not implemented since that is not
 // how plugin.EntrySchema objects are meant to be used.
 func (s EntrySchema) MarshalJSON() ([]byte, error) {
+	if s.entry == nil {
```

Comment:
What caused this change?

Reply:
We used to use the private `entrySchema`, which is just a plain-old-data (POD in C++ parlance) struct. We would serialize that as part of the graph. I started using `EntrySchema` instead, but we only fill in the parts from `entrySchema`. So this method would hit a null pointer when we tried to marshal it because `entry` (and `graph`) weren't set.

I'm open to suggestions. I could probably mess with public/private nesting more to have a public POD class we can use for external but hide it from `EntrySchema`'s docs.

Reply:
I'm still confused. I guess the specific scenario that led to `s.entry == nil` would help.

Reply:
Pushed up a separate commit that shows what I proposed instead of adding that check.

Reply:
Oh, I think I see what's going on. For external plugins, the graph nodes are `EntrySchema` objects but `s.entry == nil` (makes sense because `s.entry` is meant for core plugin entry schemas). `s.entry == nil` means those nodes just contain `entrySchema` fields, so call `entrySchema#MarshalJSON` which is what you're doing here.

That makes sense. I prefer a comment along those lines instead of the `EntrySchemaData` type b/c I don't think the comment is that confusing, while the data type is.

Reply:
Ok. I'll revert it and clarify the comment.

Reply:
Having EntrySchema and EntrySchemaData was also in danger of doing the wrong type cast, this is a lot clearer.